mantle/cmd/kolet/kolet.go

```diff
@@ -80,10 +98,108 @@ func registerTestMap(m map[string]*register.Test) {
 	}
 }
 
+// dispatchRunExtUnit returns true if unit completed successfully, false if
+// it's still running (or unit was terminated by SIGTERM)
+func dispatchRunExtUnit(unitname string, sdconn *systemddbus.Conn) (bool, error) {
+	props, err := sdconn.GetAllProperties(unitname)
+	if err != nil {
+		return false, errors.Wrapf(err, "listing unit properties")
+	}
+
+	result := props["Result"]
+	if result == "exit-code" {
+		return false, fmt.Errorf("Unit %s exited with code %d", unitname, props["ExecMainStatus"])
+	}
+
+	state := props["ActiveState"]
+	substate := props["SubState"]
+
+	switch state {
+	case "inactive":
+		fmt.Printf("Starting %s\n", unitname)
+		sdconn.StartUnit(unitname, "fail", nil)
+		return false, nil
+	case "activating":
+		return false, nil
+	case "active":
+		{
+			switch substate {
+			case "exited":
+				maincode := props["ExecMainCode"]
+				mainstatus := props["ExecMainStatus"]
+				switch maincode {
+				case CLD_EXITED:
+					if mainstatus == int32(0) {
+						return true, nil
+					} else {
+						// I don't think this can happen, we'd have exit-code above; but just
+						// for completeness
+						return false, fmt.Errorf("Unit %s failed with code %d", unitname, mainstatus)
+					}
+				case CLD_KILLED:
+					// SIGTERM; we explicitly allow that, expecting we're rebooting.
+					if mainstatus == 15 {
+						return false, nil
```

Comment:
Might be worth printing something in this case to help debugging if somehow we end with SIGTERM, but we're *not* rebooting.